blob/s3blob/s3blob.go

```diff
@@ -120,32 +125,82 @@ const Scheme = "s3"
 
 // OpenBucketURL opens the S3 bucket with the same name as the host in the URL.
 func (o *URLOpener) OpenBucketURL(ctx context.Context, u *url.URL) (*blob.Bucket, error) {
-	for k := range u.Query() {
-		return nil, fmt.Errorf("open S3 bucket %q: unknown query parameter %s", u, k)
+	opts, err := o.forParams(ctx, u.Query())
+	if err != nil {
+		return nil, fmt.Errorf("open bucket %v: %v", u, err)
+	}
+	return OpenBucket(ctx, o.ConfigProvider, u.Host, opts)
+}
+
+var legalQueryParam = map[string]bool{
+	"region":           true,
+	"endpoint":         true,
+	"disableSSL":       true,
+	"s3ForcePathStyle": true,
+}
+
+func (o *URLOpener) forParams(ctx context.Context, q url.Values) (*Options, error) {
+	for k := range q {
+		if !legalQueryParam[k] {
+			return nil, fmt.Errorf("unknown S3 query parameter %s", k)
+		}
 	}
-	return OpenBucket(ctx, o.ConfigProvider, u.Host, &o.Options)
+	var cfg aws.Config
+	override := false
+	if region := q["region"]; len(region) > 0 {
+		cfg.Region = aws.String(region[0])
+		override = true
+	}
+	if endpoint := q["endpoint"]; len(endpoint) > 0 {
```

Comment:
After careful studying of these parameters, I don't see a need yet for any other than `region`. Is there a specific use case that you are targeting? `disableSSL` in particular seems dangerous.

Reply:
These were [added by an external contributor](https://github.com/google/go-cloud/pull/1095). "these three are the ones I needed in order to use the package with a local minio instance". Apparently there are other service providers that have an S3-compatible API.

Reply:
Gotcha... what do you think about this: `Endpoint` and `DisableSSL` are somewhat redundant with each other: we could parse `Endpoint` as a URL and if it uses "http", then set `DisableSSL` automatically. If `Endpoint` is set at all, `s3ForcePathStyle` also needs to be set.

Reply:
I think it's simpler if we just map directly onto what `aws.Config` provides. I think we'll just shoot ourselves in the foot and confuse our users if we start doing non-obvious things.

Reply:
Discussed offline. I agree with the code as-is.